index.js

```diff
@@ -197,3 +211,13 @@ module.exports = function(input, map) {
   lint(input, config, this)
   this.callback(null, input, map)
 }
+
+function safeWriteCache(cachePath, cacheJson) {
+  if (fs.existsSync(cachePath)) {
+    fs.writeFileSync(cachePath, cacheJson)
+  } 
+  else {
+    fs.mkdirSync(path.dirname(cachePath))
```

Comment:
It is odd that we're using two different ways to create a directory: `mkdirSync` and `findCacheDir` (which internally [uses `mkdirp`](https://github.com/avajs/find-cache-dir/blob/7df8ba743347945d1f6d51a281fe6776d656b104/index.js#L23)).

Can you use `mkdirp` here too? You also don't need these `existsSync` checks because `mkdirp` already handles all of that.